### ISSUE TITLE
ENH : benchmark against sklearn and celer

### ISSUE THREAD

**Issue by mathurinm** [mathurinm/andersoncd#49]
A priori we should get timings similar to celer, and improve upon sklearn. 
With my current benchmark (https://github.com/mathurinm/benchmark_lasso/tree/anderson) we are slower than both : 

Some possible reasons : 
- [ ] numba is slower than cython
- [ ] using numba classes is slower > test with a harcoded lasso solver
- [ ] kkt as stopping crit overoptimizes on subproblems
- [ ] the problems in benchopt are too easy (all less than 1 s) > test on larger dataset

Feel free to edit/add stuff @QB3

**Comment by agramfort** [comment]
do you actually simulate sparse data?

**Comment by mathurinm** [comment]
Yes it's benchopt's make_correlated_data : https://benchopt.github.io/generated/benchopt.datasets.simulated.make_correlated_data.html#benchopt.datasets.simulated.make_correlated_data

**Comment by agramfort** [comment]
tu as pris un w_true sparse?

>

**Comment by mathurinm** [comment]
Approximately, benchopt's density is 0.2 so w_true has 1_000 nnz

I just pushed https://github.com/mathurinm/andersoncd/blob/master/andersoncd/tests/benchmark.py   to play faster with the code, so far we are never faster than sklearn while celer is when alpha becomes small (alpha_max / 100).

Sometimes it gets quite bad: 

################################################################################
Setup:
    X: (2000, 10000)
    alpha_max / 100
    nnz in sol: 1933
    nnz in tru: 1000
us: 24.3889 s, kkt: 3.26e-05, obj: 29.4207366036
sk: 8.0396 s, kkt: 1.70e-04, obj: 29.4213767252
cr: 5.9640 s, kkt: 9.63e-05, obj: 29.4209978211
